File: CS/Modules.TeamList/TeamListModule.cs
using Prism.Modularity;
using Prism.Regions;

namespace Modules.TeamList {
    public class TeamListModule : IModule {
        private readonly IRegionManager regionManager;
        public TeamListModule(IRegionManager regionManager) {
            this.regionManager = regionManager;
        }
        void IModule.Initialize() {
            regionManager.RegisterViewWithRegion("TeamListRegion", typeof(TeamListView));
        }
    }
}

File: VB/Modules.TeamList/TeamListModule.vb
Imports Prism.Modularity
Imports Prism.Regions

Namespace Modules.TeamList
    Public Class TeamListModule
        Implements IModule
        Private ReadOnly regionManager As IRegionManager
        Public Sub New(ByVal regionManager As IRegionManager)
            Me.regionManager = regionManager
        End Sub
        Public Sub Initialize() Implements IModule.Initialize
            regionManager.RegisterViewWithRegion("TeamListRegion", GetType(TeamListView))
        End Sub
    End Class
End Namespace
